src/Quartz/Impl/Triggers/CronTriggerImpl.cs

```diff
@@ -386,6 +392,7 @@ public CronTriggerImpl(string name, string group, string cronExpression) : base(
         string cronExpression,
         TimeZoneInfo timeZone) : base(name, group, jobName, jobGroup)
     {
+        timeProvider = TimeProvider.System;
```

Comment:
should the widest constructor take `TimeProvider? timeProvider = null` and then we default to system?

Reply:
Ok reworked, also trying to constructor chain where possible.

Reply:
A more general question overall,  as perhaps I am going about things wrong.  We can inject/pass in a TimeProvider, but there is also this SystemTime with Funcs for Now/UtcNow.      I see these as both conflicting.  

`SystemTime` being static can be set at a global level, good for objects that sit outside of DI etc.  But also being static, are there situations where there is a need for separate `TimeProviders` ?  In which case I would assume the goal is to reduce dependency on `SystemTime` class.    I can't think of a scenario where different TimeProviders would be required?

So just wondering if taking parameters for a TimeProvider actually makes sense and should methods to leaning on `SystemTime` more?